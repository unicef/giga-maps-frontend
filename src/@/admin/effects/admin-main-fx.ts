import { createEffect } from "effector"

import { createRequestAuthFx } from "~/core/auth/effects/common.fx"

import { ApiConfig, InvalidateCache } from "../types/giga-layer.type"

export const getAppConfigValuesFx = createEffect(() => {
  return createRequestAuthFx({
    url: `accounts/app_configs/`
  }) as Promise<ApiConfig>
})

export const getInvalidateCacheFx = createEffect((data: object) => {
  return createRequestAuthFx({
    url: `accounts/invalidate-cache-patterns/`,
    method: 'DELETE',
    data
  }) as Promise<InvalidateCache>
})