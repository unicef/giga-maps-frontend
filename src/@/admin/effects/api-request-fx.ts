import { createEffect } from "effector";

import { APIListType } from "~/api/types";
import { createRequestAuthFx } from "~/core/auth/effects/common.fx";

import { ApiKeysAdminRequestType } from "../types/api-request.type";

export const getCountryApiFx = createEffect((query: string) => {
  return createRequestAuthFx({
    url: `locations/countries/${query}`,
  }) as Promise<ApiKeysAdminRequestType[]>
})

export const getAllApiKeyRequest = createEffect(({ page, pageSize, countryIds, search }: { page: number; pageSize: number, countryIds?: string, search?: string }) => {
  return createRequestAuthFx({
    url: `accounts/api_keys/?page_size=${pageSize}&page=${page}${countryIds ? `&country_id__in=${countryIds}` : ''}${search ? `&search=${search}` : ''}&expand=user,api&ordering=-status`
  }) as Promise<APIListType<ApiKeysAdminRequestType>>
})

export const changeApiKeyRequestFx = createEffect(({ body, id }: { body: any; id: number }) => {
  return createRequestAuthFx({
    url: `accounts/api_keys/${id}/`,
    data: body,
    method: 'PUT'
  }) as Promise<ApiKeysAdminRequestType>
})

export const deleteApiKeyRequestFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `accounts/api_keys/${id}/`,
    method: 'DELETE'
  }) as Promise<ApiKeysAdminRequestType>
})

export const updateApiKeyExtensionFx = createEffect(({ id, body }: { id: number, body: any; }) => {
  return createRequestAuthFx({
    url: `accounts/api_keys/${id}/`,
    method: 'PUT',
    data: body
  }) as Promise<ApiKeysAdminRequestType>
});