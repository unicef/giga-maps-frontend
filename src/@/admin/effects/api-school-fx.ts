import { createEffect } from "effector"

import { APIListType } from "~/api/types"
import { createRequestAuthFx } from "~/core/auth/effects/common.fx"

import { AdminType, CsvImport, SchoolDailyType, SchoolSummaryType, SchoolType } from "../types/school.type";



export const getSchoolListFx = createEffect(({ page, pageSize, search, filter }: { page?: number; pageSize?: number; search?: string, filter?: number[] }) => {
  let url = `locations/schools/school/?`
  url += `page=${page}&page_size=${pageSize}`;

  if (filter && filter.length > 0) {
    url += `&country_id__in=${filter.join(',')}`;
  }

  if (search) {
    url += `&search=${search}`;
  }

  return createRequestAuthFx({
    url: url
  }) as Promise<APIListType<SchoolType>>
})

export const createOrUpdateSchoolFx = createEffect(({ body, isEditMode, params }: any) => {
  return createRequestAuthFx({
    method: isEditMode ? 'PUT' : 'POST',
    url: `locations/schools/school/${params.id ? params.id + '/' : ''}`,
    data: body
  }) as Promise<SchoolType>
})

export const getSchoolIdFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `locations/schools/school/${id + '/'}`
  }) as Promise<SchoolType>
})

export const deleteSchoolFx = createEffect(({ body, }: any) => {
  return createRequestAuthFx({
    method: 'DELETE',
    url: `locations/schools/school/`,
    data: body
  }) as Promise<SchoolType>
})


export const getSchoolSummaryListFx = createEffect(({ page, pageSize, search, filter, schoolId }: { page?: number; pageSize?: number; search?: string, filter?: number[], schoolId?: number }) => {
  return createRequestAuthFx({
    url: `/statistics/schoolweeklystatus/?page=${page}&page_size=${pageSize}${(filter && filter.length > 0) ? `&country_id=${filter}` : ''}${search ? `&search=${search}` : ''}${schoolId ? `&school_id=${schoolId}` : ''}`
  }) as Promise<APIListType<SchoolSummaryType>>
})

export const getSchoolSummaryIdFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `/statistics/schoolweeklystatus/${id + '/'}`
  }) as Promise<SchoolSummaryType>
})

export const deleteSchoolSummaryFx = createEffect(({ body, }: any) => {
  return createRequestAuthFx({
    method: 'DELETE',
    url: `/statistics/schoolweeklystatus/`,
    data: body
  }) as Promise<SchoolType>
})

export const createOrUpdateSchoolSummaryFx = createEffect(({ body, isEditMode, params }: any) => {
  return createRequestAuthFx({
    method: isEditMode ? 'PUT' : 'POST',
    url: `/statistics/schoolweeklystatus/${params.id ? params.id + '/' : ''}`,
    data: body
  }) as Promise<SchoolType>
})

export const getSchoolDailyListFx = createEffect(({ page, pageSize, search, filter }: { page?: number; pageSize?: number; search?: string, filter?: number[] }) => {
  return createRequestAuthFx({
    url: `/statistics/schooldailystatus/?page=${page}&page_size=${pageSize}${(filter && filter.length > 0) ? `&country_id=${filter}` : ''}${search ? `&search=${search}` : ''}`
  }) as Promise<APIListType<SchoolDailyType>>
})

export const getSchoolDailyIdFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `/statistics/schooldailystatus/${id + '/'}`
  }) as Promise<SchoolSummaryType>
})

export const deleteSchoolDailyFx = createEffect(({ body, }: any) => {
  return createRequestAuthFx({
    method: 'DELETE',
    url: `statistics/schooldailystatus/`,
    data: body
  }) as Promise<SchoolSummaryType>
})

export const createOrUpdateSchoolDailyFx = createEffect(({ body, isEditMode, params }: any) => {
  return createRequestAuthFx({
    method: isEditMode ? 'PUT' : 'POST',
    url: `statistics/schooldailystatus/${params.id ? params.id + '/' : ''}`,
    data: body
  }) as Promise<SchoolType>
})


export const importCsvFx = createEffect(({ formData }: any) => {
  return createRequestAuthFx({
    method: 'POST',
    url: `/locations/schools/fileimport/`,
    body: formData
  }) as Promise<CsvImport>
})

export const getCsvImportListFx = createEffect(({ page, pageSize, search }: { page?: number; pageSize?: number; search?: string }) => {
  return createRequestAuthFx({
    url: `locations/schools/fileimport/?page=${page}&page_size=${pageSize}${search ? `&search=${search}` : ''}`
  }) as Promise<APIListType<CsvImport>>
})

export const getAdmin1ListFx = createEffect(({ layerName, countryId }: { layerName?: string; countryId?: string; }) => {
  return createRequestAuthFx({
    url: `locations/country-admin-metadata/?layer_name=${layerName}&country_id=${countryId}&ordering=description,name&expand=country,parent&page_size=500&fields=id,name,description,name_en&page=1`
  }) as Promise<APIListType<AdminType>>
})


export const getAdmin2ListFx = createEffect(({ layerName, countryId, parentId }: { layerName?: string; countryId?: string; parentId?: number }) => {
  return createRequestAuthFx({
    url: `locations/country-admin-metadata/?layer_name=${layerName}&country_id=${countryId}&ordering=description,name&expand=country,parent${parentId ? `&parent_id=${parentId}` : ''}&page_size=500&fields=id,name,description,name_en&page=1`
  }) as Promise<APIListType<AdminType>>
})