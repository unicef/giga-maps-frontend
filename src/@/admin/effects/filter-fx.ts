import { createEffect } from "effector"
import { createRequestAuthFx } from "~/core/auth/effects/common.fx"
import { APIListType } from "~/api/types"
import { FilterConfiguration, FilterListType } from "../types/filter-list.type"
import { FilterAllValueType } from "../types/filter-list-type"

export const getFilterListFx = createEffect(({ page, pageSize, search }: { page: number, pageSize: number, search?: string }) => {
  const query = new URLSearchParams()
  query.set('page_size', String(pageSize))
  query.set('page', String(page))
  if (search) {
    query.set('search', search)
  }
  return createRequestAuthFx({
    url: `accounts/adv_filters/?${query.toString()}&ordering=-last_modified_at&expand=column_configuration,published_by`
  }) as Promise<APIListType<FilterListType>>
})

export const getFilterPublishedListFx = createEffect(() => {
  return createRequestAuthFx({
    url: `accounts/adv_filters/?status=PUBLISHED`
  }) as Promise<APIListType<FilterListType>>
})

export const filterColumnListFx = createEffect(() => {
  return createRequestAuthFx({
    url: `accounts/column_configurations/?is_filter_applicable=true`
  }) as Promise<APIListType<FilterConfiguration>>
})

export const deleteFilterFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `accounts/adv_filters/${id}/`,
    method: 'DELETE'
  }) as Promise<FilterListType>
})

export const addFilterFx = createEffect(({ body }: { body: FilterAllValueType }) => {
  return createRequestAuthFx({
    url: `accounts/adv_filters/`,
    method: 'POST',
    data: body
  }) as Promise<FilterListType>
})
export const editFilterFx = createEffect(({ id, body }: { id: number, body: FilterAllValueType | { status: string } }) => {
  return createRequestAuthFx({
    url: `accounts/adv_filters/${id + '/'}`,
    method: 'PUT',
    data: body
  }) as Promise<FilterListType>
})

export const getFilterListIdFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `accounts/adv_filters/?id=${id}`
  }) as Promise<APIListType<FilterListType>>
})

export const publishFilterFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    method: 'PUT',
    url: `accounts/adv_filters/${id}/publish/`,
  }) as Promise<FilterListType>
})