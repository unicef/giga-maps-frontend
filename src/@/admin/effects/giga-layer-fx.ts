import { attach, createEffect } from "effector"

import { APIListType } from "~/api/types"
import { createRequestAuthFx } from "~/core/auth/effects/common.fx"

import { DataLayer, DataSource, LayerStatusType, PreviewDataType } from "../types/giga-layer.type"

export const commonLayerListFx = createEffect(({ page, pageSize, search }: { page: number, pageSize: number; search?: string }) => {
  return createRequestAuthFx({
    url: `accounts/layers/?expand=created_by,last_modified_by,published_by&page_size=${pageSize}&page=${page}${search ? `&search=${search}` : ''}&ordering=-last_modified_at,name`
  }) as Promise<APIListType<DataLayer>>
})

export const cacheDataLayerListFx = attach({
  effect: commonLayerListFx,
  mapParams: (props) => props,
})

export const getDataLayerListFx = attach({
  effect: commonLayerListFx,
  mapParams: (props) => props,
})


export const getDataLayerByIdFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `accounts/layers/?id=${id}&expand=created_by,last_modified_by,published_by`
  }) as Promise<APIListType<DataLayer>>
})

export const createDataLayerFx = createEffect(({ body, isEditMode, params }: any) => {
  return createRequestAuthFx({
    method: isEditMode ? 'PUT' : 'POST',
    url: `accounts/layers/${params.id ? params.id + '/' : ''}`,
    data: body
  }) as Promise<DataLayer>
})

export const getApiSourceValuesFx = createEffect(({ type }: { type: string }) => {
  return createRequestAuthFx({
    url: `accounts/data_sources/?data_source_type__in=${type}&status=PUBLISHED`
  }) as Promise<APIListType<DataSource>>
})

export const publishDataLayerFx = createEffect(({ id, status }: { id: number, status: LayerStatusType }) => {
  return createRequestAuthFx({
    method: 'PUT',
    data: { status },
    url: `accounts/layers/${id}/publish/`,
  }) as Promise<DataLayer>
})

export const getDataPreviewFx = createEffect(({ id }: { id: number }) => {
  return createRequestAuthFx({
    url: `/accounts/layers/${id}/preview/`,
  }) as Promise<PreviewDataType>
})
