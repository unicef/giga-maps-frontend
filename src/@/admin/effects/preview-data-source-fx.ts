import { createEffect } from "effector";
import mapboxgl, { Map } from "mapbox-gl";

import { PreviewDataType } from "../types/giga-layer.type";
import { addCountryBoundries } from "./map-country-boundary.fx";
import { DEFAULT_SOURCE } from "~/@/map/map.constant";

const getSchoolsGeoJson = (data: PreviewDataType['map']) => {
  return {
    type: 'FeatureCollection',
    features: data.map((item: PreviewDataType['map'][0]) => ({
      type: 'Feature',
      properties: {
        description:
          `<strong>${item.school_name}</strong>
        <p>Giga id: ${item.school_id_giga}</p>
        <p>Lat: ${item.latitude}, Lon: ${item.longitude}</p>`
      },
      geometry: {
        'type': 'Point',
        'coordinates': [item.longitude, item.latitude]
      },
      id: item.id,
    })),
  };
};


export const plotMapData = (map: Map, mapData: PreviewDataType['map']) => {
  const data = getSchoolsGeoJson(mapData) as unknown as GeoJSON.Geometry;
  map.addSource(DEFAULT_SOURCE, {
    type: 'geojson',
    data
  });

  map.addLayer({
    'id': 'places',
    'type': 'circle',
    'source': DEFAULT_SOURCE,
    'paint': {
      'circle-color': '#4264fb',
      'circle-radius': 6,
      'circle-stroke-width': 2,
      'circle-stroke-color': '#ffffff'
    }
  });

  const popup = new mapboxgl.Popup({
    closeButton: true,
    closeOnClick: true,
    className: 'preview-data-popup'
  });

  const createPopup = (e: any) => {
    const coordinates = e.features[0].geometry.coordinates.slice();
    const description = e.features[0].properties.description;

    // Ensure that if the map is zoomed out such that multiple
    // copies of the feature are visible, the popup appears
    // over the copy being pointed to.
    while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
      coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
    }

    // Populate the popup and set its coordinates
    // based on the feature found.
    popup.setLngLat(coordinates).setHTML(`<div class="map-content-popup">${description}</div>`).addTo(map);
  }

  map.on('mouseenter', 'places', (e) => {
    // Change the cursor style as a UI indicator.
    map.getCanvas().style.cursor = 'pointer';
    // Copy coordinates array.
    createPopup(e);
  })
    .on('click', 'places', (e) => createPopup(e))

  map.on('mouseleave', 'places', () => { });
}

export const previewDataSourceFx = createEffect(({ map, mapData }: { map: Map, mapData: PreviewDataType['map'] }) => {
  if (!map) return;
  try {
    addCountryBoundries(map);
    plotMapData(map, mapData);
  } catch (e) {
    console.log(e);
  }
})
