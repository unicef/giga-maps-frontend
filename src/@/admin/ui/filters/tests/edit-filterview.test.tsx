import { useStore } from "effector-react";
import EditFilterList from "../edit-filter.view";
import { render } from "@testing-library/react";
import { adminFilterRoute, editAdminFilter, router } from "~/core/routes";
import { getFilterListIdFx } from "~/@/admin/effects/filter-fx";

describe('EditFilterList', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should navigate to adminFilterRoute if id is not provided', () => {
    editAdminFilter.navigate({ id: 0 })
    const { asFragment } = render(<EditFilterList />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call getFilterListIdFx with id when id is provided', () => {
    editAdminFilter.navigate({ id: 1 })
    const { asFragment } = render(<EditFilterList />);
    expect(asFragment()).toMatchSnapshot();
  });

});