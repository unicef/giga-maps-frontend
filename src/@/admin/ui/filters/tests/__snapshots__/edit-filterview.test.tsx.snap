// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFilterList should call getFilterListIdFx with id when id is provided 1`] = `
<DocumentFragment>
  .c0 {
  height: 100vh;
  width: 18.5rem;
  z-index: 999;
  position: fixed;
  right: 0;
  top: 0;
  background: #222;
}

.c1 {
  max-height: calc(100vh - 3.1rem);
}

.c2 {
  padding: 0rem 1rem;
}

.c3 {
  display: flex;
  justify-content: space-between;
}

.c3 h6 {
  font-size: 0.875rem;
  color: #f5f5f5;
  padding-top: 1rem;
}

.c6 {
  color: #c0c0c0;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  font-size: 0.75rem;
}

.c6.center-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c6 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance::before {
  background-color: #c0c0c0;
}

.c6 .cds--checkbox-label::before,
.c6 .cds--radio-button__appearance,
.c6 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance {
  border-color: #c0c0c0;
}

.c6 .cds--label {
  color: #c0c0c0;
}

.c6 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon path:not([data-icon-path]):not([fill=none]),
.c6 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon {
  fill: #c0c0c0;
}

.c8 {
  margin-top: 0.5rem;
}

.c8 .cds--label {
  color: #c0c0c0;
}

.c8 .cds--select-input {
  background: #333333;
  color: #f5f5f5;
}

.c8 .cds--list-box__label {
  color: #f5f5f5;
}

.c8 svg {
  fill: #f5f5f5;
}

.c7 .cds--text-input {
  background: #333333;
  color: #f5f5f5;
}

.c7 .cds--text-input::placeholder {
  color: #f5f5f5;
}

.c7 .cds--text-input:disabled {
  color: #f5f5f5;
  -webkit-text-fill-color: #f5f5f5;
  opacity: 0.4;
}

.c7 input:-webkit-autofill,
.c7 input:-webkit-autofill:hover,
.c7 input:-webkit-autofill:focus,
.c7 textarea:-webkit-autofill,
.c7 textarea:-webkit-autofill:hover,
.c7 textarea:-webkit-autofill:focus {
  box-shadow: 0 0 0 1000px #333333 inset;
  -webkit-text-fill-color: #f5f5f5;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: center;
}

.c5 {
  width: 90%;
}

.c9 Button {
  width: 50%;
}

<div
    class="c0"
  >
    <form
      class="cds--form"
      style="display: flex; flex-direction: column; color: white; height: inherit; justify-content: space-between;"
    >
      <div
        class="scrollbar-container c1 ps"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h6>
              Edit filter
            </h6>
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="0 c5"
          >
            <div
              class="c6"
            >
              Unique Code
            </div>
            <div
              class="cds--form-item c7 cds--text-input-wrapper"
            >
              <div
                class="cds--text-input__label-wrapper"
              />
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input"
                    disabled=""
                    id="layer-code"
                    name="code"
                    placeholder="Enter layer unique code"
                    required=""
                    title="Enter layer unique code"
                    type="text"
                    value="TEST-FILTER-004"
                  />
                  <span
                    class="cds--text-input__counter-alert"
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="cds--form-item c8"
            >
              <div
                class="cds--select"
              >
                <label
                  class="cds--label"
                  for="parameter-select"
                >
                  Parameter
                </label>
                <div
                  class="cds--select-input__wrapper"
                >
                  <select
                    class="cds--select-input"
                    id="parameter-select"
                    name="column_configuration"
                    placeholder="Choose parameter type"
                    required=""
                    title=""
                  >
                    <option
                      class="cds--select-option"
                      value=""
                    >
                      Choose parameter
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="cds--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="cds--form-item c8"
            >
              <div
                class="cds--select"
              >
                <label
                  class="cds--label"
                  for="filter-type-select"
                >
                  Filter type
                </label>
                <div
                  class="cds--select-input__wrapper"
                >
                  <select
                    class="cds--select-input"
                    id="filter-type-select"
                    name="type"
                    placeholder="Choose filter type"
                    required=""
                    title=""
                  >
                    <option
                      class="cds--select-option"
                      value=""
                    >
                      Choose filter type
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="cds--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="c6"
            >
              Filter name
            </div>
            <div
              class="cds--form-item c7 cds--text-input-wrapper"
            >
              <div
                class="cds--text-input__label-wrapper"
              />
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input"
                    id="filter-name"
                    name="name"
                    placeholder="Enter filter name"
                    required=""
                    title="Enter filter name"
                    type="text"
                    value="Available devices"
                  />
                  <span
                    class="cds--text-input__counter-alert"
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="c6"
            >
              Filter description
            </div>
            <div
              class="cds--form-item c7 cds--text-input-wrapper"
            >
              <div
                class="cds--text-input__label-wrapper"
              />
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input"
                    id="filter-description"
                    name="description"
                    placeholder="Enter description"
                    title="Enter description"
                    type="text"
                    value="Filter to check the device availability"
                  />
                  <span
                    class="cds--text-input__counter-alert"
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="cds--form-item c8"
            >
              <div
                class="cds--select"
              >
                <label
                  class="cds--label"
                  for="filter-type-select"
                >
                  Query Param filter
                </label>
                <div
                  class="cds--select-input__wrapper"
                >
                  <select
                    class="cds--select-input"
                    id="filter-type-select"
                    name="query_param_filter"
                    placeholder="Choose query param filter"
                    required=""
                    title=""
                  >
                    <option
                      class="cds--select-option"
                      value=""
                    >
                      Choose query param
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="cds--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ps__rail-x"
          style="left: 0px; top: 0px;"
        >
          <div
            class="ps__thumb-x"
            style="left: 0px; width: 0px;"
            tabindex="0"
          />
        </div>
        <div
          class="ps__rail-y"
          style="top: 0px; left: 0px;"
        >
          <div
            class="ps__thumb-y"
            style="top: 0px; height: 0px;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="1 c9"
      >
        <button
          class="cds--btn cds--btn--primary"
          style="background: rgb(71, 71, 71);"
          type="button"
        >
          Cancel
        </button>
        <button
          class="cds--btn cds--btn--primary cds--btn--disabled"
          disabled=""
          type="submit"
        >
          Update
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`EditFilterList should navigate to adminFilterRoute if id is not provided 1`] = `
<DocumentFragment>
  .c0 {
  height: 100vh;
  width: 18.5rem;
  z-index: 999;
  position: fixed;
  right: 0;
  top: 0;
  background: #222;
}

.c1 {
  max-height: calc(100vh - 3.1rem);
}

.c2 {
  padding: 0rem 1rem;
}

.c3 {
  display: flex;
  justify-content: space-between;
}

.c3 h6 {
  font-size: 0.875rem;
  color: #f5f5f5;
  padding-top: 1rem;
}

.c6 {
  color: #c0c0c0;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  font-size: 0.75rem;
}

.c6.center-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c6 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance::before {
  background-color: #c0c0c0;
}

.c6 .cds--checkbox-label::before,
.c6 .cds--radio-button__appearance,
.c6 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance {
  border-color: #c0c0c0;
}

.c6 .cds--label {
  color: #c0c0c0;
}

.c6 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon path:not([data-icon-path]):not([fill=none]),
.c6 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon {
  fill: #c0c0c0;
}

.c8 {
  margin-top: 0.5rem;
}

.c8 .cds--label {
  color: #c0c0c0;
}

.c8 .cds--select-input {
  background: #333333;
  color: #f5f5f5;
}

.c8 .cds--list-box__label {
  color: #f5f5f5;
}

.c8 svg {
  fill: #f5f5f5;
}

.c7 .cds--text-input {
  background: #333333;
  color: #f5f5f5;
}

.c7 .cds--text-input::placeholder {
  color: #f5f5f5;
}

.c7 .cds--text-input:disabled {
  color: #f5f5f5;
  -webkit-text-fill-color: #f5f5f5;
  opacity: 0.4;
}

.c7 input:-webkit-autofill,
.c7 input:-webkit-autofill:hover,
.c7 input:-webkit-autofill:focus,
.c7 textarea:-webkit-autofill,
.c7 textarea:-webkit-autofill:hover,
.c7 textarea:-webkit-autofill:focus {
  box-shadow: 0 0 0 1000px #333333 inset;
  -webkit-text-fill-color: #f5f5f5;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: center;
}

.c5 {
  width: 90%;
}

.c9 Button {
  width: 50%;
}

<div
    class="c0"
  >
    <form
      class="cds--form"
      style="display: flex; flex-direction: column; color: white; height: inherit; justify-content: space-between;"
    >
      <div
        class="scrollbar-container c1 ps"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h6>
              Edit filter
            </h6>
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="0 c5"
          >
            <div
              class="c6"
            >
              Unique Code
            </div>
            <div
              class="cds--form-item c7 cds--text-input-wrapper"
            >
              <div
                class="cds--text-input__label-wrapper"
              />
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input"
                    disabled=""
                    id="layer-code"
                    name="code"
                    placeholder="Enter layer unique code"
                    required=""
                    title="Enter layer unique code"
                    type="text"
                    value=""
                  />
                  <span
                    class="cds--text-input__counter-alert"
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="cds--form-item c8"
            >
              <div
                class="cds--select"
              >
                <label
                  class="cds--label"
                  for="parameter-select"
                >
                  Parameter
                </label>
                <div
                  class="cds--select-input__wrapper"
                >
                  <select
                    class="cds--select-input"
                    id="parameter-select"
                    name="column_configuration"
                    placeholder="Choose parameter type"
                    required=""
                    title=""
                  >
                    <option
                      class="cds--select-option"
                      value=""
                    >
                      Choose parameter
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="cds--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="cds--form-item c8"
            >
              <div
                class="cds--select"
              >
                <label
                  class="cds--label"
                  for="filter-type-select"
                >
                  Filter type
                </label>
                <div
                  class="cds--select-input__wrapper"
                >
                  <select
                    class="cds--select-input"
                    id="filter-type-select"
                    name="type"
                    placeholder="Choose filter type"
                    required=""
                    title=""
                  >
                    <option
                      class="cds--select-option"
                      value=""
                    >
                      Choose filter type
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="cds--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="c6"
            >
              Filter name
            </div>
            <div
              class="cds--form-item c7 cds--text-input-wrapper"
            >
              <div
                class="cds--text-input__label-wrapper"
              />
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input"
                    id="filter-name"
                    name="name"
                    placeholder="Enter filter name"
                    required=""
                    title="Enter filter name"
                    type="text"
                    value=""
                  />
                  <span
                    class="cds--text-input__counter-alert"
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="c6"
            >
              Filter description
            </div>
            <div
              class="cds--form-item c7 cds--text-input-wrapper"
            >
              <div
                class="cds--text-input__label-wrapper"
              />
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input"
                    id="filter-description"
                    name="description"
                    placeholder="Enter description"
                    title="Enter description"
                    type="text"
                    value=""
                  />
                  <span
                    class="cds--text-input__counter-alert"
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="cds--form-item c8"
            >
              <div
                class="cds--select"
              >
                <label
                  class="cds--label"
                  for="filter-type-select"
                >
                  Query Param filter
                </label>
                <div
                  class="cds--select-input__wrapper"
                >
                  <select
                    class="cds--select-input"
                    id="filter-type-select"
                    name="query_param_filter"
                    placeholder="Choose query param filter"
                    required=""
                    title=""
                  >
                    <option
                      class="cds--select-option"
                      value=""
                    >
                      Choose query param
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="cds--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ps__rail-x"
          style="left: 0px; top: 0px;"
        >
          <div
            class="ps__thumb-x"
            style="left: 0px; width: 0px;"
            tabindex="0"
          />
        </div>
        <div
          class="ps__rail-y"
          style="top: 0px; left: 0px;"
        >
          <div
            class="ps__thumb-y"
            style="top: 0px; height: 0px;"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="1 c9"
      >
        <button
          class="cds--btn cds--btn--primary"
          style="background: rgb(71, 71, 71);"
          type="button"
        >
          Cancel
        </button>
        <button
          class="cds--btn cds--btn--primary cds--btn--disabled"
          disabled=""
          type="submit"
        >
          Update
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
